spec/ddtrace/contrib/rails/rails_semantic_logger_auto_injection_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'ddtrace/contrib/rails/rails_helper'
+
+RSpec.describe 'Rails Log Auto Injection' do
+  include Rack::Test::Methods
+  include_context 'Rails test application'
+
+  let(:routes) do
+    {
+      '/semantic_logger' => 'semantic_logger_test#index'
+    }
+  end
+
+  let(:controllers) do
+    [
+      semantic_logger_controller
+    ]
+  end
+
+  let(:semantic_logger_controller) do
+    stub_const('SemanticLoggerTestController', Class.new(ActionController::Base) do
+      def index
+        Rails.logger.info('MINASWAN')
+        render inline: '<html> <head> </head> <body> <div> Hello from index </div> </body> </html>'
+      end
+    end)
+  end
+
+  before do
+    Datadog.configuration[:rails].reset_options!
+    Datadog.configure do |c|
+      c.use :rails, log_injection: true
+    end
+
+    allow(ENV).to receive(:[]).and_call_original
+  end
+
+  after do
+    Datadog.configuration[:rails].reset_options!
+  end
+
+  context 'with Log_Injection Enabled' do
+    # defined in rails support apps
+    let(:logs) { log_output.string }
+
+    if Rails.version >= '4.0'
+      context 'with Semantic Logger' do
```

Comment:
You can clean this up a bit by using the `if:` argument for RSpec: https://github.com/DataDog/dd-trace-rb/blob/db89b55870f193e3fc94e5dc7695756d5cf85438/spec/ddtrace_integration_spec.rb#L120